### ISSUE TITLE
Compiling with JDK 8 results in java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;

### ISSUE THREAD

**Issue by nickheniser** [vy/log4j2-logstash-layout#69]
I am compiling my program with a JDK 8 compiler, and seeing java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer; at runtime.

I am bringing in `implementation "com.vlkan.log4j2:log4j2-logstash-layout:1.0.4"`

Maybe it has something to do with this: https://stackoverflow.com/questions/61267495/exception-in-thread-main-java-lang-nosuchmethoderror-java-nio-bytebuffer-flip

Is there something I am doing incorrectly?

```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="debug">
	<Appenders>
		<Console name="JsonConsole" target="SYSTEM_OUT">
			<LogstashLayout dateTimeFormatPattern="yyyy-MM-dd'T'HH:mm:ss.SSSZ" 
			eventTemplateUri="classpath:LogstashFedex.json" 
			prettyPrintEnabled="true" 
			stackTraceEnabled="true"
			timeZoneId="GMT"/>
		</Console>
		
	</Appenders>
	<Loggers>


		<Root level="INFO">
			<AppenderRef ref="JsonConsole" />
		</Root>
	</Loggers>
</Configuration>
```
```
2020-10-29 10:41:54,025 main DEBUG Initializing configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml]
2020-10-29 10:41:54,034 main DEBUG Installed script engines
2020-10-29 10:41:54,287 main DEBUG Oracle Nashorn Version: 1.8.0_211, Language: ECMAScript, Threading: Not Thread Safe, Compile: true, Names: {nashorn, Nashorn, js, JS, JavaScript, javascript, ECMAScript, ecmascript}
2020-10-29 10:41:54,288 main DEBUG PluginManager 'Core' found 114 plugins
2020-10-29 10:41:54,288 main DEBUG PluginManager 'Level' found 0 plugins
2020-10-29 10:41:54,291 main DEBUG PluginManager 'Lookup' found 13 plugins
2020-10-29 10:41:54,292 main DEBUG Building Plugin[name=layout, class=com.vlkan.log4j2.logstash.layout.LogstashLayout].
2020-10-29 10:41:54,308 main DEBUG PluginManager 'TypeConverter' found 26 plugins
2020-10-29 10:41:54,319 main DEBUG LogstashLayout$Builder(Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), prettyPrintEnabled="true", locationInfoEnabled="null", stackTraceEnabled="true", emptyPropertyExclusionEnabled="null", dateTimeFormatPattern="yyyy-MM-dd'T'HH:mm:ss.SSSZ", timeZoneId="GMT", locale="null", eventTemplate="null", eventTemplateUri="classpath:LogstashFedex.json", EventTemplateAdditionalFields=null, stackTraceElementTemplate="null", stackTraceElementTemplateUri="null", mdcKeyPattern="null", ndcPattern="null", lineSeparator="null", maxByteCount="null", maxStringLength="null", objectMapperFactoryMethod="null", mapMessageFormatterIgnored="null")
2020-10-29 10:41:54,554 main DEBUG Building Plugin[name=appender, class=org.apache.logging.log4j.core.appender.ConsoleAppender].
2020-10-29 10:41:54,562 main DEBUG ConsoleAppender$Builder(target="SYSTEM_OUT", follow="null", direct="null", bufferedIo="null", bufferSize="null", immediateFlush="null", ignoreExceptions="null", LogstashLayout(com.vlkan.log4j2.logstash.layout.LogstashLayout@747f281), name="JsonConsole", Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:54,564 main DEBUG Jansi is not installed, cannot find org.fusesource.jansi.WindowsAnsiOutputStream
2020-10-29 10:41:54,564 main DEBUG Starting OutputStreamManager SYSTEM_OUT.false.false
2020-10-29 10:41:54,564 main DEBUG Building Plugin[name=appenders, class=org.apache.logging.log4j.core.config.AppendersPlugin].
2020-10-29 10:41:54,566 main DEBUG createAppenders(={JsonConsole})
2020-10-29 10:41:54,567 main DEBUG Building Plugin[name=AppenderRef, class=org.apache.logging.log4j.core.config.AppenderRef].
2020-10-29 10:41:54,571 main DEBUG createAppenderRef(ref="JsonConsole", level="null", Filter=null)
2020-10-29 10:41:54,581 main DEBUG Building Plugin[name=logger, class=org.apache.logging.log4j.core.config.LoggerConfig].
2020-10-29 10:41:54,583 main DEBUG createLogger(additivity="false", level="INFO", name="CRSV_STATS", includeLocation="null", ={JsonConsole}, ={}, Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:54,586 main DEBUG Building Plugin[name=AppenderRef, class=org.apache.logging.log4j.core.config.AppenderRef].
2020-10-29 10:41:54,587 main DEBUG createAppenderRef(ref="JsonConsole", level="null", Filter=null)
2020-10-29 10:41:54,587 main DEBUG Building Plugin[name=root, class=org.apache.logging.log4j.core.config.LoggerConfig$RootLogger].
2020-10-29 10:41:54,588 main DEBUG createLogger(additivity="null", level="INFO", includeLocation="null", ={JsonConsole}, ={}, Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:54,588 main DEBUG Building Plugin[name=loggers, class=org.apache.logging.log4j.core.config.LoggersPlugin].
2020-10-29 10:41:54,589 main DEBUG createLoggers(={CRSV_STATS, root})
2020-10-29 10:41:54,590 main DEBUG Configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] initialized
2020-10-29 10:41:54,590 main DEBUG Starting configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml]
2020-10-29 10:41:54,590 main DEBUG Started configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] OK.
2020-10-29 10:41:54,591 main DEBUG Shutting down OutputStreamManager SYSTEM_OUT.false.false-1
2020-10-29 10:41:54,591 main DEBUG Shut down OutputStreamManager SYSTEM_OUT.false.false-1, all resources released: true
2020-10-29 10:41:54,591 main DEBUG Appender DefaultConsole-1 stopped with status true
2020-10-29 10:41:54,591 main DEBUG Stopped org.apache.logging.log4j.core.config.DefaultConfiguration@c818063 OK
2020-10-29 10:41:54,635 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93
2020-10-29 10:41:54,638 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=StatusLogger
2020-10-29 10:41:54,640 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=ContextSelector
2020-10-29 10:41:54,642 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Loggers,name=CRSV_STATS
2020-10-29 10:41:54,643 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Loggers,name=
2020-10-29 10:41:54,644 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Appenders,name=JsonConsole
2020-10-29 10:41:54,648 main DEBUG Reconfiguration complete for context[name=73d16e93] at URI C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml (org.apache.logging.log4j.core.LoggerContext@12299890) with optional ClassLoader: null
2020-10-29 10:41:54,649 main DEBUG Shutdown hook enabled. Registering a new one.
2020-10-29 10:41:54,651 main DEBUG LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890] started OK.
2020-10-29 10:41:57,401 main DEBUG Using configurationFactory org.apache.logging.log4j.core.config.ConfigurationFactory$Factory@2e54db99
2020-10-29 10:41:57,403 main INFO Log4j appears to be running in a Servlet environment, but there's no log4j-web module available. If you want better web container support, please add the log4j-web JAR to your web archive or server lib directory.
2020-10-29 10:41:57,405 main DEBUG Loaded configuration from C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml
2020-10-29 10:41:57,405 main DEBUG Starting LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890] with configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml]...
2020-10-29 10:41:57,406 main DEBUG Initializing configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml]
2020-10-29 10:41:57,407 main DEBUG Installed script engines
2020-10-29 10:41:57,411 main DEBUG Oracle Nashorn Version: 1.8.0_211, Language: ECMAScript, Threading: Not Thread Safe, Compile: true, Names: {nashorn, Nashorn, js, JS, JavaScript, javascript, ECMAScript, ecmascript}
2020-10-29 10:41:57,411 main DEBUG PluginManager 'Core' found 114 plugins
2020-10-29 10:41:57,412 main DEBUG PluginManager 'Level' found 0 plugins
2020-10-29 10:41:57,412 main DEBUG PluginManager 'Lookup' found 13 plugins
2020-10-29 10:41:57,412 main DEBUG Building Plugin[name=layout, class=com.vlkan.log4j2.logstash.layout.LogstashLayout].
2020-10-29 10:41:57,413 main DEBUG LogstashLayout$Builder(Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), prettyPrintEnabled="true", locationInfoEnabled="null", stackTraceEnabled="true", emptyPropertyExclusionEnabled="null", dateTimeFormatPattern="yyyy-MM-dd'T'HH:mm:ss.SSSZ", timeZoneId="GMT", locale="null", eventTemplate="null", eventTemplateUri="classpath:LogstashFedex.json", EventTemplateAdditionalFields=null, stackTraceElementTemplate="null", stackTraceElementTemplateUri="null", mdcKeyPattern="null", ndcPattern="null", lineSeparator="null", maxByteCount="null", maxStringLength="null", objectMapperFactoryMethod="null", mapMessageFormatterIgnored="null")
2020-10-29 10:41:57,415 main DEBUG Building Plugin[name=appender, class=org.apache.logging.log4j.core.appender.ConsoleAppender].
2020-10-29 10:41:57,415 main DEBUG ConsoleAppender$Builder(target="SYSTEM_OUT", follow="null", direct="null", bufferedIo="null", bufferSize="null", immediateFlush="null", ignoreExceptions="null", LogstashLayout(com.vlkan.log4j2.logstash.layout.LogstashLayout@773cbf4f), name="JsonConsole", Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:57,417 main DEBUG Jansi is not installed, cannot find org.fusesource.jansi.WindowsAnsiOutputStream
2020-10-29 10:41:57,417 main DEBUG Building Plugin[name=appenders, class=org.apache.logging.log4j.core.config.AppendersPlugin].
2020-10-29 10:41:57,417 main DEBUG createAppenders(={JsonConsole})
2020-10-29 10:41:57,417 main DEBUG Building Plugin[name=AppenderRef, class=org.apache.logging.log4j.core.config.AppenderRef].
2020-10-29 10:41:57,418 main DEBUG createAppenderRef(ref="JsonConsole", level="null", Filter=null)
2020-10-29 10:41:57,418 main DEBUG Building Plugin[name=logger, class=org.apache.logging.log4j.core.config.LoggerConfig].
2020-10-29 10:41:57,418 main DEBUG createLogger(additivity="false", level="INFO", name="CRSV_STATS", includeLocation="null", ={JsonConsole}, ={}, Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:57,419 main DEBUG Building Plugin[name=AppenderRef, class=org.apache.logging.log4j.core.config.AppenderRef].
2020-10-29 10:41:57,419 main DEBUG createAppenderRef(ref="JsonConsole", level="null", Filter=null)
2020-10-29 10:41:57,419 main DEBUG Building Plugin[name=root, class=org.apache.logging.log4j.core.config.LoggerConfig$RootLogger].
2020-10-29 10:41:57,419 main DEBUG createLogger(additivity="null", level="INFO", includeLocation="null", ={JsonConsole}, ={}, Configuration(C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml), Filter=null)
2020-10-29 10:41:57,420 main DEBUG Building Plugin[name=loggers, class=org.apache.logging.log4j.core.config.LoggersPlugin].
2020-10-29 10:41:57,420 main DEBUG createLoggers(={CRSV_STATS, root})
2020-10-29 10:41:57,420 main DEBUG Configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] initialized
2020-10-29 10:41:57,420 main DEBUG Starting configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml]
2020-10-29 10:41:57,420 main DEBUG Started configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] OK.
2020-10-29 10:41:57,421 main DEBUG Appender JsonConsole stopped with status true
2020-10-29 10:41:57,421 main DEBUG Stopped XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] OK
2020-10-29 10:41:57,422 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93
2020-10-29 10:41:57,422 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=StatusLogger
2020-10-29 10:41:57,422 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=ContextSelector
2020-10-29 10:41:57,423 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Loggers,name=CRSV_STATS
2020-10-29 10:41:57,423 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Loggers,name=
2020-10-29 10:41:57,423 main DEBUG Registering MBean org.apache.logging.log4j2:type=73d16e93,component=Appenders,name=JsonConsole
2020-10-29 10:41:57,423 main DEBUG LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890] started OK with configuration XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml].
2020-10-29 10:41:58,002 main DEBUG AsyncLogger.ThreadNameStrategy=CACHED
Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
	at com.vlkan.log4j2.logstash.layout.LogstashLayout.encode(LogstashLayout.java:223)
	at com.vlkan.log4j2.logstash.layout.LogstashLayout.encode(LogstashLayout.java:57)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.directEncodeEvent(AbstractOutputStreamAppender.java:177)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.tryAppend(AbstractOutputStreamAppender.java:170)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:161)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:448)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:433)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:417)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:403)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:63)
	at org.apache.logging.log4j.core.Logger.logMessage(Logger.java:146)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2091)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:1971)
	at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1829)
	at org.apache.logging.log4j.jcl.Log4jLog.error(Log4jLog.java:116)
	at org.springframework.boot.SpringApplication.reportFailure(SpringApplication.java:826)
	at org.springframework.boot.SpringApplication.handleRunFailure(SpringApplication.java:801)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
	at com.fedex.nxgen.crs.v17.cloud.Application.main(Application.java:14)
2020-10-29 10:41:58,034 pool-1-thread-1 DEBUG Stopping LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890]
2020-10-29 10:41:58,035 pool-1-thread-1 DEBUG Stopping LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890]...
2020-10-29 10:41:58,036 pool-1-thread-1 DEBUG Shutting down OutputStreamManager SYSTEM_OUT.false.false
2020-10-29 10:41:58,036 pool-1-thread-1 DEBUG Shut down OutputStreamManager SYSTEM_OUT.false.false, all resources released: true
2020-10-29 10:41:58,036 pool-1-thread-1 DEBUG Appender JsonConsole stopped with status true
2020-10-29 10:41:58,036 pool-1-thread-1 DEBUG Stopped XmlConfiguration[location=C:\code\CRSV\crsv-fast-cloud\bin\main\log4j2.xml] OK
2020-10-29 10:41:58,036 pool-1-thread-1 DEBUG Stopped LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@12299890] with status true
```

**Comment by vy** [comment]
I will see what I can do about it. I think I need to use [animal-sniffer-maven-plugin](https://www.mojohaus.org/animal-sniffer/animal-sniffer-maven-plugin/index.html).

I see that you are using an old version of JVM 8, i.e., 1.8.0_211. Is it an option for you to upgrade to the latest Java 8?

**Comment by nickheniser** [comment]
Thanks!  I'll see if updating my java version will help, didn't realize they would update the signature in later version of java8?  I'll report back with findings.

**Comment by vy** [comment]
@nickheniser, I have just released `1.0.5`, this should have fixed your issue. Would you mind checking, please? (Don't hesitate to reopen the issue if your problem is not solved.)

**Comment by nickheniser** [comment]
> @nickheniser, I have just released `1.0.5`, this should have fixed your issue. Would you mind checking, please? (Don't hesitate to reopen the issue if your problem is not solved.)

Fixed for me!  Thanks for the quick help!
